mobile/lib/shared/services/api.service.dart

```diff
@@ -70,6 +74,29 @@ class ApiService {
     return url;
   }
 
+  Future<void> _isEndpointAvailable(String serverUrl) async {
```

Comment:
Before trying the endpoint, we can check the network state if there is any network at all.

Reply:
Currently there is a bug that always returns `none` for Connectivity on the iOS simulator. If I add the check the endpoint will always be "unavailable" and causing the app to not work on the iOS simulator.

https://github.com/fluttercommunity/plus_plugins/issues/198

Reply:
Nvm I added a work around for this